servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/versionstore/ConfigurableVersionStoreFactory.java

```diff
@@ -133,7 +108,7 @@ private VersionStore persistVersionStore() {
 
   private VersionStore databaseAdapterVersionStore() {
     try {
-      DatabaseAdapter da = databaseAdapter.select().get();
+      DatabaseAdapter da = databaseAdapter.select(WIthInitializedRepository.Literal.INSTANCE).get();
```

Comment:
This doesn't look required. I suppose it's just an attempt to create a symmetry with how the new-model `Persist` instance is created and for correctness, since, after all, the old-model bean performs repo initialization.

Reply:
Sadly not only for symmetry. Without the literal it barks "I got two beans!" :(

Reply:
The Nessie-Quarkus-CLI tool comes into play here as well.

Reply:
Ah true, now we have the `ObservingDatabaseAdapter` bean, so a qualifier is required to disambiguate 👍